database/database.go

```diff
@@ -153,6 +154,49 @@ func (db DatastoreBacked) StateForDomain(domain string) (state DomainState, err
 	return state, nil
 }
 
+// StatesForDomains returns the domains states for the given domains
+func (db DatastoreBacked) StatesForDomains(domains []string) (states []DomainState, err error) {
+	// Set up datastore context
+	c, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	client, datastoreErr := db.backend.NewClient(c, db.projectID)
+	if datastoreErr != nil {
+		return nil, datastoreErr
+	}
+
+	getDomainStates := func(keys []*datastore.Key) ([]DomainState, error) {
+		domainStates := make([]DomainState, len(keys))
+		getErr := client.GetMulti(c, keys, domainStates)
+		if getErr != nil {
+			return nil, getErr
+		}
+		for i := range domainStates {
+			domainStates[i].Name = keys[i].Name
+		}
+		return domainStates, nil
+	}
+
+	var keys []*datastore.Key
+	var res []DomainState
+	var getErr error
```

Comment:
same comment about using err here

Reply:
(and I think you can omit this declaration since the error is always used only in the block it is received).

Reply:
Done